transformation/src/api/rest/pipelineConfigEndpoint.ts

```diff
@@ -45,102 +47,90 @@ export class PipelineConfigEndpoint {
-    const config = req.body as PipelineConfig
-    if (!config.transformation) {
-      config.transformation = { func: 'return data;' }
+    const validator = new PipelineConfigDTOValidator()
+    if (!validator.validate(req.body)) {
+      res.status(400).json({ errors: validator.getErrors() })
+      return
     }
+    const config = req.body
     const savedConfig = await this.pipelineConfigManager.create(config)
-    res.setHeader('Content-Type', 'application/json')
-    res.setHeader('location', `/configs/${savedConfig.id}`)
-    res.writeHead(201)
-    res.write(JSON.stringify(savedConfig))
-    res.end()
-  }
-
-  getByDatasourceId = async (datasourceId: number, req: express.Request, res: express.Response): Promise<void> => {
-    const configs = await this.pipelineConfigManager.getByDatasourceId(datasourceId)
-    res.setHeader('Content-Type', 'application/json')
-    res.writeHead(200)
-    res.write(JSON.stringify(configs))
-    res.end()
+    res.location(`/configs/${savedConfig.id}`).status(201).json(savedConfig)
   }
 
   getOne = async (req: express.Request, res: express.Response): Promise<void> => {
-    const configId = req.params.id
-    if (!configId) {
-      res.writeHead(400)
-      res.write('Path parameter id is missing or is incorrect')
-      res.end()
+    const configId = parseInt(req.params.id)
+    if (Number.isNaN(configId)) {
+      res.status(400).send('Path parameter id is missing or is incorrect')
       return
     }
-    const config = await this.pipelineConfigManager.get(+configId)
+    const config = await this.pipelineConfigManager.get(configId)
     if (!config) {
       res.status(404).send('Config not found')
+    } else {
+      res.status(200).json(config)
     }
-    res.setHeader('Content-Type', 'application/json')
-    res.writeHead(200)
-    res.write(JSON.stringify(config))
-    res.end()
   }
 
   getAll = async (req: express.Request, res: express.Response): Promise<void> => {
-    const datasourceId = req.query.datasourceId
-    if (datasourceId) {
-      return this.getByDatasourceId(+datasourceId, req, res)
+    const datasourceId = parseInt(this.getQueryParameter(req, 'datasourceId'))
+    let configs
+    if (Number.isNaN(datasourceId)) {
+      configs = await this.pipelineConfigManager.getAll()
+    } else {
+      configs = await this.pipelineConfigManager.getByDatasourceId(datasourceId)
     }
+    res.status(200).json(configs)
+  }
 
-    const configs = await this.pipelineConfigManager.getAll()
-    res.setHeader('Content-Type', 'application/json')
-    res.writeHead(200)
-    res.write(JSON.stringify(configs))
-    res.end()
+  private getQueryParameter (req: express.Request, key: string): string {
+    const value = req.query[key]
+    if (typeof value === 'string') {
+      return value
+    } else if (Array.isArray(value) && typeof value[0] === 'string') {
+      return value[0]
+    }
+    return ''
```

Comment:
You could improve the readability of the function by removing the else statement.
E.g.
```js
if (typeof value === 'string') {
      return value
} 

if (Array.isArray(value) && typeof value[0] === 'string') {
      return value[0]
}

return ''
```

Reply:
And you could also refactor the `typeof` into a own method to improve the readability.

```js
if (isString(value)) {...} 
if (Array.isArray(value) && isString(value[0])) 
```